### ISSUE TITLE
rewrite structure can't pull db settings from environment variables

### ISSUE THREAD

**Issue by harrissAvalon** [wp-cli/wp-cli#3640]
My wp-config.php file is configured to pull the MySQL database settings from environmental variables. However when I'm trying to run `wp rewrite structure` I keep receiving the following error because it's not reading those environment variables:

```
Error establishing a database connection. This either means that the username and password information in your `wp-config.php` file is incorrect or we can’t contact the database server at `mysql`. This could mean your host’s database server is down.
```

I tested this by hard-coding the settings and running the rewrite structure and it was able to read then.

**Comment by danielbachhuber** [comment]
> However when I'm trying to run `wp rewrite structure` I keep receiving the following error because it's not reading those environment variables:

This is because `wp rewrite structure` uses `proc_open()` to launch a new process, and the custom environmental variables aren't passed through to the child process.

Ideas on how this could be resolved?

**Comment by danielbachhuber** [comment]
Closing this as wontfix.

When I originally implemented `WP_CLI::runcommand()`, I looked into passing all environment variables through to the new process. However, it doesn't appear possible for a PHP process to know of all of its environment variables, only to fetch the values for specific variables. `WP_CLI::runcommand()` persists the environment variables it knows of (a160df6bb933d09b5504403e640d1b445c7934c3) but there's no way for it to persist environment variables it doesn't know of.
